demos/benney_luke/benney_luke.py.rst

```diff
@@ -0,0 +1,198 @@
+    & \qquad \qquad + \mu\left( \nabla\delta q \cdot\nabla\phi - \frac{3}{2}q\,\delta q  \right) \,dx\,dy\,dt,
+
+where the spatial domain is assumed to be :math:`\Omega` with natural boundary conditions, namely Neumann conditions on all the boundaries. In addition, suitable end-point conditions at :math:`t=0` and :math:`t=T` are used. Note that the introduction of the auxiliary function :math:`q` is performed in order to lower the highest derivatives. This is advantageous in a :math:`C^0` finite element formulation and motivated the modification of the "standard" Benney-Luke equations. The partial variations in the last line of the variational principle can be integrated by parts in order to get expressions that only depend on :math:`\delta\eta,\,\delta\phi,\,\delta q` and not their derivatives:
+
+.. math::
+
+  0 = \int_0^T \int_{\Omega} \left( \phi_t - \frac{\mu}{2}\Delta\phi_t + \eta + \frac{\epsilon}{2}\left|\nabla\phi\right|^2 \right)\delta\eta
+                              - \left( \eta_t - \frac{\mu}{2}\Delta\eta_t + \nabla\cdot\bigl(\left(1+\epsilon\eta\right)\!\nabla\phi\bigr)+\mu\Delta q \right)\delta\phi
+                              - \mu\left( \Delta\phi + \frac{3}{2}q \right)\delta q \,dx\,dy\,dt.
+
+Since the variations :math:`\delta\eta,\,\delta\phi,\,\delta q` are arbitrary, the modified Benney-Luke equations then arise for functions :math:`\eta,\phi,q\in V` from a suitable function space :math:`V` and are given by:
+
+.. math::
+
+  \phi_t - \frac{\mu}{2}\Delta\phi_t + \eta + \frac{\epsilon}{2}\left|\nabla\phi\right|^2 &= 0 \\
+  \eta_t - \frac{\mu}{2}\Delta\eta_t + \nabla\cdot\bigl(\left(1+\epsilon\eta\right)\!\nabla\phi\bigr)+\mu\Delta q &= 0 \\
+  q &= - \frac{2}{3}\Delta\phi.
+
+We can either directly use the partial variations in the variational principle above (last line) as the fundamental weak formulation (with :math:`\delta\phi,\, \delta\eta,\, \delta q` playing the role of test functions), or multiply the equations by a test function :math:`v\in V` and integrate over the domain in order to obtain a weak formulation in a classic manner
+
+.. math::
+
+  \int_{\Omega} \phi_t\,v + \frac{\mu}{2}\nabla\phi_t\cdot\nabla v + \eta\,v + \frac{\epsilon}{2}\nabla\phi\cdot\nabla\phi\,v \,dx\,dy &= 0 \\
+  \int_{\Omega} \eta_t\,v + \frac{\mu}{2}\nabla\eta_t\cdot\nabla v - \left(1+\epsilon\eta\right)\nabla\phi\cdot\nabla v - \mu\nabla q\cdot\nabla v \,dx\,dy &= 0 \\
+  \int_{\Omega} q\,v - \frac{2}{3}\nabla\phi\cdot\nabla v \,dx\,dy &= 0.
+
+Note that the Neumann boundary conditions have been used to remove every surface term that resulted from the integration by parts. Moreover, the variational form of the system requires the use of a symplectic integrator for the time-discretisation. Here we choose the 2nd-order Stormer-Verlet scheme, which requires two half-steps to update :math:`\phi` in time (one implicit and one explicit in general) and one (implicit) step for :math:`\eta`:

```

Comment:
What is the best way to give references in demos? For the Stormer-Verlet it would be "E. Hairer, C. Lubich, G. Wanner (2006) Geometric numerical integration. Springer." and for the KdV "P.G. Drazin, R.S. Johnson (1989) Solitons: an Introduction. Cambridge University Press."